### PR TITLE
classes are now SourceWithPlotting

### PR DESCRIPTION
**Pull request by shimwell**
This PR adds some plotting features to the package.

The sources are now instances of ```openmc_source_plotter.SourceWithPlotting()``` which is just a class that inherits ```openmc.Source``` and adds some plotting methods.

The additional plotting methods are ```.plot_source_position```, ```.plot_source_energy```, ```.plot_source_direction``` and there is also a sampling method called ```sample_initial_particles```

This doesn't remove any of the existing plotting methods but adds a few new ones.

### CODE REVIEW

**Comment by shimwell** [comment]
Looks like the ```tests/test_tokamak_source.py::test_strengths_are_normalised``` is failing but I think that might be unrelated to this PR

**Comment by RemDelaporteMathurin** [comment]
> Looks like the `tests/test_tokamak_source.py::test_strengths_are_normalised` is failing but I think that might be unrelated to this PR

The tests pass on develop so it has to be related to this PR

File: README.md [hunk 1]
```diff
@@ -43,6 +43,8 @@ my_source = TokamakSource(
     shafranov_factor=0.44789,
     triangularity=0.270,
     ion_temperature_beta=6
+    sample_size=1000,
+    angles = (0, 2*3.14)  # angle in radians
```
Comment (RemDelaporteMathurin) [inline README.md:47]:
```suggestion
    angles=(0, 2*3.14)  # angles in radians
```

**Review by RemDelaporteMathurin — COMMENTED** [review]
If I'm being totally honest, I prefered the former way of plotting with functions rather than methods.

It's more "out-of-the-box" and worked with any source. Also, if I understood it correctly, it was more compatible with openmc in the sense that if openmc users wanted to plot an openmc.Source object, they could just use the appropriate openmc-source-plotter function.
Now, users need to remember to create a osp.SourceWithPlotting object which induces backwards changes.

In a nutshell, i prefered the old way of using osp functions 😄

**Comment by shimwell** [comment]
ok fair enough, I shall refactor and come back with an improved version. I was perhaps a bit lazy here.
